core/settings/qPref_private.h

```diff
@@ -0,0 +1,72 @@
+// SPDX-License-Identifier: GPL-2.0
+#ifndef QPREFPRIVATE_H
+#define QPREFPRIVATE_H
+
+// Header used by all qPref<class> implementations to avoid duplicating code
+
+#include <QObject>
+#include <QSettings>
+#include <QVariant>
+#include "core/qthelper.h"
+
+#define COPY_TXT(name, string) \
+{ \
+	free((void *)prefs.name); \
+	prefs.name = copy_qstring(string); \
+}
+
+#define LOADSYNC_INT(name, field) \
+{ \
+	QSettings s; \
+	if (doSync) \
+		s.setValue(group + name, prefs.field); \
+	else \
+		prefs.field = s.value(group + name, default_prefs.field).toInt(); \
+}
```

Comment:
> Is there a reason to create a new object every time?

If you look at #1423, these macros seem to replace a full function body. Which then raises the question, why not generate the whole function with a macro?

I think these macros should go in a PR where their use is seen. Otherwise it's really difficult to judge their usefulness.

Reply:
Agree they should not be in this PR.

Moreover, even though they replace a function body... they replace a getter, that is also a setter.  and called by the setter, to actually set/save values. This does not make sense to me.

The setter code is always the same anyway, is just one line and can be made more clear by doing just that.

Reply:
It looks like this comment is still unresolved?

Reply:
I still would prefer if these macros were introduced with a PR that also introduces their use. Otherwise it's hard to get the "vision". Note: not necessarily the same commit, but the same PR would be nice.